two_factor/forms.py

```diff
@@ -176,3 +177,31 @@ def clean(self):
 
 class BackupTokenForm(AuthenticationTokenForm):
     otp_token = forms.CharField(label=_("Token"))
+
+
+class TokenAuthenticationForm(AuthenticationForm):
+    user_id = forms.CharField()
+    backend = forms.CharField()
+
+    def __init__(self, *args, **kwargs):
+        super(TokenAuthenticationForm, self).__init__(*args, **kwargs)
+        del self.fields['username']
+        del self.fields['password']
+
+    def authenticate(self):
+        UserModel = get_user_model()
+        user = UserModel.objects.get(id=self.cleaned_data['user_id'])
```

Comment:
Hmm that's a good point, but you asked me to *only* add the security fix related code in this PR, so I guess it could be an improvement for later as this was already the case before. 
+ The hacker would also have to access the user's computer which would still be on the token form page as I don't think there is any way to access the token page directly. 
Also, accessing the auth page clears the session info.
What do you think?